File: LeetCode/2342-max-sum-of-a-pair-with-equal-sum-of-digits/2342-max-sum-of-a-pair-with-equal-sum-of-digits.cpp
class Solution {
public:
    int sumOfDigit(int n) {
        int sum = 0;
        while (n > 0) {
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }
    int maximumSum(vector<int>& nums) {
        unordered_map<int, int> mp;
        int n = nums.size();
        sort(nums.begin(), nums.end());
        int ans = 0;
        for (int i = 0; i < n; i++) {
            int sum = sumOfDigit(nums[i]);
            if (mp.find(sum) != mp.end()) {
                ans = max(ans, nums[i] + nums[mp[sum]]);
            }
            mp[sum] = i;
        }
        return ans == 0 ? -1 : ans;
    }
};

File: LeetCode/2342-max-sum-of-a-pair-with-equal-sum-of-digits/2342-max-sum-of-a-pair-with-equal-sum-of-digits.py
class Solution:
    def sum_of_digit(self, n: int) -> int:
        return sum(int(digit) for digit in str(n))
    
    def maximumSum(self, nums: List[int]) -> int:
        nums.sort()
        digit_sum_map = {}
        ans = -1

        for num in nums:
            s = self.sum_of_digit(num)
            if s in digit_sum_map:
                ans = max(ans, num + digit_sum_map[s])
            digit_sum_map[s] = num
        
        return ans